### PR TITLE
core:archive when looking for other climulti processes, also make sure idSite is same

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

As title. This is in case there are multiple archivers running on the same machine, we can pick up climulti requests for other sites.

### Review

* [ ] Functional review done
* [ ] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] Code review done
* [ ] Tests were added if useful/possible
* [ ] Reviewed for breaking changes
* [ ] Developer changelog updated if needed
* [ ] Documentation added if needed
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by diosmosis** [comment]
FYI @tsteur

File: core/CronArchive/QueueConsumer.php [hunk 1]
```diff
@@ -397,18 +397,25 @@ private function shouldSkipArchiveBecauseLowerPeriodOrSegmentIsInProgress(array
                 continue;
             }
 
+            if (empty($archiveBeingProcessed['idSite'])
```
Comment (sgiehl) [inline core/CronArchive/QueueConsumer.php:400]: Haven't had a deeper look at the code, but should it actually happen that `idSite` isn't set for an archiving process? If so wouldn't we need to check if `idSite` of both aren't empty here?  🤔

Comment (diosmosis) [reply core/CronArchive/QueueConsumer.php:400]: It's a bit confusing, but here `$archiveBeingProcessed` are the parameters for a climulti:request, while `$archiveToProcess` is a row in archive_invalidations. So `idSite` might not be set in a `climulti:request` execution, but `idsite` will always be in `$archiveToProcess`.

**Comment by tsteur** [comment]
Seems tests are passing @diosmosis . is it good to merge?

**Pull request merged**